pkg/registry/apis/datasource/handler.go

```diff
@@ -0,0 +1,17 @@
+package datasource
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"k8s.io/apiserver/pkg/registry/rest"
+)
+
+type HTTPRequestHandlerFunc func(ctx context.Context, pluginCtx backend.PluginContext, responder rest.Responder) (http.Handler, error)
+
+type PluginRequestHandlers struct {
```

Comment:
🤔 why not https://github.com/grafana/grafana-plugin-sdk-go/blob/main/backend/datasource/serve.go#L8 

IIUC, it achieves the same thing -- and keeps the core vs external distinction irrelevant for anyone using it.

Reply:
If we use the SDK contracts here, then for a response for an external plugin over gRPC will go:

`[plugin gRPC server]` -> encode as Arrow (backend to proto)  -> `[plugin gRPC client]` -> decode Arrow (proto to backend) -> `[gRPC server]` -> encode as Arrow (backend to proto) -> `[gRPC client]` -> decode Arrow (proto to backend)

As opposed to:

`[plugin gRPC server]` -> encode as Arrow (backend to proto) -> `[plugin gRPC client]` -> passthru -> `[gRPC server]` -> passthru -> `[gRPC client]` -> decode Arrow (proto to backend)

Reply:
I understand that for the protobuf side... but not for HTTP

Reply:
Yes exactly - I made a note in the description that this doesn't need to be this way for HTTP, but for now it seemed more straightforward to consolidate both paths into one. If that's a deal breaker, then we can look at trying to keep the structure as is but throwing the client proto stuff in on top?